code/mob.dm

```diff
@@ -485,6 +485,10 @@
 	if (illumplane) //Wire: Fix for Cannot modify null.alpha
 		illumplane.alpha = 255
 
+	if(HAS_MOB_PROPERTY(src, PROP_PROTANOPIA))
+		var/list/protanopia_matrix = list(MATRIX_PROTANOPIA)
+		src.client?.color = list(protanopia_matrix)
```

Comment:
Just do src.client?.color = list(MATRIX_PROTAMOPIA). Here your are nesting it too see deeply.

Reply:
Done.